internal/server/singleprocess/service_deploy.go

```diff
@@ -55,7 +55,7 @@ func (s *service) UpsertDeployment(
 		}
 
 		// Create a context that will timeout relatively quickly
-		ctx, cancel := context.WithTimeout(ctx, 5*time.Second)
+		ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
```

Comment:
This is an unrelated change that we want to get in as well to increase the buffer horizon has to setup the hostname.